src/AggregateFunctions/DDSketch/Store.h

```diff
@@ -228,13 +228,9 @@ class DDSketchDenseStore : public DDSketchStore
         else
         {
             UInt32 new_length = getNewLength(new_min_key, new_max_key);
-            if (new_length > bins.capacity())
-            {
-                bins.reserve(new_length);
-            }
-            std::size_t old_size = bins.size();
             bins.resize(new_length);
-            std::fill(bins.begin() + old_size, bins.end(), 0.0);
+            bins.resize(bins.capacity());
+            new_max_key = new_min_key + int(bins.size()) - 1;
```

Comment:
Wait, this looks wrong, `new_max_key` should be the max observed key, independent of the array size, no?

Reply:
Right, it was incorrect. I reverted it back.

This suggestion was behaving weird for relative accuracy such as 0.0001. I am not sure yet. I will be looking into it.
```
bins.resize(new_length);

// (At this point bins.capacity() is usually greater than bins.size(), because resize() grows capacity() exponentially.
//  such that a series of increasing resize()s takes O(final_size) time, even if resizing by one element at a time.
//  But here we do an additional O(n) pass in adjust(), so exponentially growing capacity() doesn't help,
//  we're still O(n^2).
//  So we make size() grow exponentially together with capacity(), by resizing again.)

bins.resize(bins.capacity());

```